frontend/src/pages/add-blog.tsx

```diff
@@ -34,12 +34,22 @@ function AddBlog() {
     isFeaturedPost: false,
   });
 
+  //checks the length of the categories array and if the category is already selected
+  const isValidCategory = (category: string): boolean => {
+    return formData.categories.length >= 3 && !formData.categories.includes(category);
+  };
+
   const handleInputChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
     const { name, value } = e.target;
     setFormData({ ...formData, [name]: value });
   };
 
   const handleCategoryClick = (category: string) => {
+    if (formData.categories.length === 3 && !formData.categories.includes(category)) {
+      toast.error('Select up to three categories.');
+      return;
+    }
+
```

Comment:
Why are we adding this? it is redundant, disabled category already takes care of this

Reply:
No, i don't think only by adding a cursor-not-allowed class alone can take care of it because even if i add this class, i can still select the category, the functionally of the click is not yet disabled, so just to disable this functionality i have added this line.

Although, i will remove the toast message.

Reply:
@vivek6201 perfect, Thank you for the value addition.